packages/core/src/shared/lsp/lspResolver.ts

```diff
@@ -90,7 +91,22 @@ export class LanguageServerResolver {
 
     /** Finds an older, cached version of the LSP server bundle. */
     private async getFallbackServer(latestVersion: LspVersion): Promise<LspResult> {
-        const fallbackDirectory = await this.getFallbackDir(latestVersion.serverVersion)
+        const cachedVersions = await this.getCachedVersions()
+        if (cachedVersions.length === 0) {
+            /**
+             * at this point the latest version doesn't exist locally, lsp download (with retries) failed, and there are no cached fallback versions.
+             * This _probably_ only happens when the user hit a firewall/proxy issue, since otherwise they would probably have at least
+             * one other language server locally
+             */
+            throw new ToolkitError(
+                `Unable to download dependencies from ${this.manifestUrl}. Check your network connectivity or firewall configuration and then try again.`,
+                {
+                    code: 'NetworkConnectivityError',
```

Comment:
these error codes auto-populate the `reason` field in telemetry right? If so, this will be important to track going forward to gauge impact. My suspicion is this only affects a very low % of users.

Reply:
Yeah exactly, that's part of why I wanted to add the code here. It should give us a rough idea of how many people are effected